packages/core/src/anchor/ethereum/ethereum-anchor-service.ts

```diff
@@ -111,6 +113,42 @@ export class EthereumAnchorService implements AnchorService {
     return [this._chainId]
   }
 
+  private _carFileFromRequestAnchorParams(params: RequestAnchorParams): CAR {
+    const carFactory = new CARFactory()
+    const car = carFactory.build()
+    // In the testing code imitate CAS logic to check that the cid in genesis matches the cid of streamID
+    const genesisBlock = new CarBlock(params.genesisCid, params.genesisBlock)
+    car.blocks.put(genesisBlock)
+    const tipBlock = new CarBlock(params.tip, params.tipBlock)
+    car.blocks.put(tipBlock)
+    if (params.genesisLinkCid) {
+      const genesisLinkBlock = new CarBlock(params.genesisLinkCid, params.genesisLinkBlock)
+      car.blocks.put(genesisLinkBlock)
+    }
+    if (params.tipLinkCid) {
+      const tupLinkBlock = new CarBlock(params.tipLinkCid, params.tipLinkBlock)
+      car.blocks.put(tupLinkBlock)
+    }
+    if (params.tipCacaoCid) {
+      const cacaoBlock = new CarBlock(params.tipCacaoCid, params.tipCacaoBlock)
+      car.blocks.put(cacaoBlock)
+    }
+
+    car.put(
+      {
+        timestamp: params.timestampISO,
+        streamId: params.streamId.toString(),
+        genesisCid: params.genesisCid.toString(),
+        genesisLinkCid: params.genesisLinkCid?.toString() ?? '',
+        tipCid: params.tip.toString(),
```

Comment:
```suggestion
        tip: params.tip,
```
IPLD supports storing CID types directly. 
It's implied that "tip" is a CID.